### PR TITLE
encrypted secret api key in the database

### PR DESCRIPTION
**Pull request by andykim**
## Description
- Encrypted secret API key in the database
- Fix Secret API Key Disclosure
https://xendit.atlassian.net/browse/TPI-8120
https://xendit.atlassian.net/browse/TPI-8121

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] General improvement (non-breaking feature that might not necessarily impact customer)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code (if necessary), particularly in hard-to-understand areas
- [ ] My changes generate no new **vulnerabilities**
- [ ] I have added tests that prove my fix is effective or that my feature works

## Merging
- [ ] I am aware which environment I intend to merge (staging | production)
- [ ] I am aware that a deployment will be triggered if I merge to staging and release branch
- [ ] I informed the team about my merge if it has risk of breaking the target API

## Deployment
- [ ] Unit tests passed
- [ ] UAT passed and communicated to QA and Product is informed about this deployment
- [ ] I'm aware off all the commits / merges behind that will be included in this deployment
- [ ] Feature flags are ready and communicated with the team

Please follow our [deployment guideline](https://xendit.atlassian.net/wiki/spaces/TPI/pages/2331476248/Engineer+Onboarding#Releasing-Flow).

## Screenshot

`put screenshot here if there are any ui changes`

### CODE REVIEW

**Review by yanoandri — APPROVED** [review]

File: xendit_pos/models/pos_payment_method.py [hunk 1]
```diff
@@ -26,10 +28,35 @@ def _get_payment_terminal_selection(self):
...
 
+    def __init__(self) -> None:
+        super().__init__()
+
+    @api.onchange('xendit_pos_secret_key')
+    def _onchange_xendit_secret_key(self):
+        if self.xendit_pos_secret_key:
+
+            # Generate the encrypt key using to encrypt secret key
+            if self.xendit_encrypt_key is False or self.xendit_encrypt_key == "":
```
Comment (ikanyu) [inline xendit_pos/models/pos_payment_method.py:44]: random style comments, let's keep to only one style. either `''` or `""`

File: xendit_pos/models/pos_payment_method.py [hunk 2]
```diff
@@ -26,10 +28,35 @@ def _get_payment_terminal_selection(self):
...
+                self.xendit_encrypt_key = encrypt.generateKey()
+
+            # Set terminal_identifier and encrypt secret key
+            self.xendit_pos_terminal_identifier = encrypt.generateKey()
+            self.xendit_pos_secret_key = encrypt.encrypt(self.xendit_pos_secret_key, self.xendit_encrypt_key)
+        else:
+            ValidationError('Invalid xendit_pos_secret_key')
```
Comment (ikanyu) [inline xendit_pos/models/pos_payment_method.py:51]: Shall we keep an enums of these validation errors?

File: xendit_pos/models/pos_payment_method.py (hunk 1 above)
Comment (andykim) [reply xendit_pos/models/pos_payment_method.py]: @ikanyu updated.

File: xendit_pos/models/pos_payment_method.py (hunk 2 above)
Comment (andykim) [reply xendit_pos/models/pos_payment_method.py:51]: @ikanyu I think no need for now.

**Review by ikanyu — APPROVED** [review]

**Pull request merged**